components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -60,20 +62,57 @@ export default function TeamSettings() {
             const billingMode = await getGitpodService().server.getBillingModeForTeam(team.id);
             setBillingMode(billingMode);
         })();
-    }, []);
+    }, [team, user]);
 
-    if (!isUserOwner) {
-        return <Redirect to="/" />;
-    }
-    const deleteTeam = async () => {
+    const updateTeamInformation = useCallback(async () => {
+        if (!team || errorMessage || !teams) {
+            return;
+        }
+        try {
+            const updatedTeam = await getGitpodService().server.updateTeam(team.id, { name: teamName });
+            const updatedTeams = [...teams?.filter((t) => t.id !== team.id)];
+            updatedTeams.push(updatedTeam);
+            setTeams(updatedTeams);
+            setUpdated(true);
+            setTimeout(() => setUpdated(false), 3000);
+        } catch (error) {
+            setErrorMessage(`Failed to update team information: ${error.message}`);
+        }
+    }, [team, errorMessage, teams, teamName, setTeams]);
+
+    const onNameChange = useCallback(
+        async (event: React.ChangeEvent<HTMLInputElement>) => {
+            if (!team) {
+                return;
+            }
+            const newName = event.target.value || "";
+            setTeamName(newName);
+            if (newName.trim().length === 0) {
+                setErrorMessage("Team name must not be empty.");
```

Comment:
**suggestion**: We should probably update the saving and feedback interaction to also match the UX on the user profile, but this is out of the scope of this PR. 🍒